### PR TITLE
Add special build mode for OSS-Fuzz

### PR DESCRIPTION
**Pull request by kevinbackhouse**
This is a follow-up to #1869. It turns out that even the `-fsanitize=fuzzer-no-link` and `-fsanitize=fuzzer` flags don't work with the OSS-Fuzz build. Instead, we have to add `$LIB_FUZZING_ENGINE` as a linker flag when we build the fuzz target. That's because OSS-Fuzz also wants to build for AFL and honggfuzz, not just libFuzzer. Each of those fuzzers uses a different `main` function wrapper around [`LLVMFuzzerTestOneInput`](https://github.com/Exiv2/exiv2/blob/f8bfad1551e1b96a673c2b129bce720d79ddd6a6/fuzz/fuzz-read-print-write.cpp#L7).

### CODE REVIEW

**Review by clanmills — APPROVED** [review]
This looks good, @kevinbackhouse 

My only request is to document these options in README.md.

**Comment by kevinbackhouse** [comment]
@clanmills: Thanks! I'll update the documentation in a follow-up PR. I want to include hyperlinks to the build script in the OSS-Fuzz repo, so it will be easier if I wait until they have merged my pull request.

**Comment by clanmills** [comment]
That's fine, @kevinbackhouse I approved it so you could push on with the OSS-Fuzz business.  Somebody else could/should take on the documentation.

**Pull request merged**
